### DISCUSSION TITLE
Troubles configuring pylsp

### DISCUSSION THREAD

**Discussion by bornedj** [VonHeikemen/lsp-zero.nvim#305] [Q&A]
Hello, not sure if this is correct place for this or if it's a potential bug with nvim-lspconfig, but I'm having troubles passing the setup to pylsp. A minimal version of my config looks like the following: 

```lua
local lsp = require('lsp-zero').preset("recommended")

lsp.ensure_installed({
    'pylsp',
   --more servers...
})

lsp.on_attach(function(client, bufnr)
    local opts = { buffer = bufnr, remap = false }
    vim.keymap.set("n", "gd", function() vim.lsp.buf.definition() end, opts)
    -- more keymaps...
end)

-- disabling pycodestyle
require('lspconfig').pylsp.setup({
    on_attach = function(client, bufnr)
        print('hello pylsp')
    end,
    settings = {
        pylsp = {
            plugins = {
                pycodestyle = {
                    enabled = false
                },
            }
        }
    }
})

lsp.setup()
```

Notably I'm unable to see anything print when using the on_attach function within the lspconfig as is, but if the `lsp.on_attach()` function is used in it's place within the pylsp config I am able to print when a python file is opened. I've taken the configuration settings from [here](https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md) and I've tried multiple other settings with no avail.

Any help is greatly appreciated!

**Comment by VonHeikemen** [comment]
Do you have this problem with other language servers?

**Reply by bornedj** [reply]
I'm only using a non default config with jdtls, which I am managing separately as java is a problem child. I can change some other server's settings to check if this issue persists.

**Reply by bornedj** [reply]
I've tested with tsserver, and rust_analyzer and I'm also not seeing a print or setting change with either. Other than the lspconfig setups, it's behaving as expected.

**Reply by VonHeikemen** [reply]
@bornedj  Have you tried with another kind of test? Like creating a keymap?

```lua
    on_attach = function(client, bufnr)
        vim.keymap.set('n', 'Q', 'echo "server works"', {buffer = bufnr})
    end,
```

**Reply by bornedj** [reply]
I've now tested this with the same servers as above, and the keymaps do not work when set in the lspconfigs setup.
